waku/v2/protocol/rln/group_manager/dynamic/dynamic.go

```diff
@@ -73,10 +73,9 @@ func (gm *DynamicGroupManager) handler(events []*contracts.RLNMemberRegistered,
 			}
 			eventsPerBlock = append(eventsPerBlock, event)
 			toInsertTable.Set(event.Raw.BlockNumber, eventsPerBlock)
-
-			if event.Raw.BlockNumber > lastBlockProcessed {
-				lastBlockProcessed = event.Raw.BlockNumber
-			}
+		}
+		if event.Raw.BlockNumber > lastBlockProcessed {
```

Comment:
Not sure this fully solves the issue. Imagine a case where blocks 1000 to 2000 are processed, but `events` is empty. In this case I think `lastBlockProcessed` will still be wrong.

I would say what matter here is
`gm.lastBlockProcessed = lastBlockProcessed`

When no events happens `latestProcessBlock` should be the latest processed block

Reply:
True, will take a relook at whole logic once.

Reply:
Addressed in [99a3510](https://github.com/waku-org/go-waku/pull/1013/commits/99a351057083ff2a7d186abcf922b9209f36a87c)